tests/postgres_tests/__init__.py

```diff
@@ -12,14 +12,14 @@ class PostgreSQLTestCase(TestCase):
     @classmethod
     def tearDownClass(cls):
         # No need to keep that signal overhead for non PostgreSQL-related tests.
-        from django.contrib.postgres.signals import register_hstore_handler
+        from django.contrib.postgres.signals import register_type_handlers
 
-        connection_created.disconnect(register_hstore_handler)
+        connection_created.disconnect(register_type_handlers)
         super().tearDownClass()
 
 
 @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific tests")
 # To locate the widget's template.
 @modify_settings(INSTALLED_APPS={'append': 'django.contrib.postgres'})
-class PostgreSQLWidgetTestCase(WidgetTest):
+class PostgreSQLWidgetTestCase(WidgetTest, PostgreSQLTestCase):
```

Comment:
This ended up being the cause of test isolation failures. Subclasses of `PostgreSQLWidgetTestCase` were installing `'django.contrib.postgres'` which was connecting the `register_type_handlers` signal without ever disconnecting it. I assume it wasn't discovered before because `psycopg2.register_hstore` was using internal connection and cursors objects bypassing `SimpleTestCase` disallowed queries checks and avoiding running connection initialization logic of `django.db.backend.postgresql`.